apps/op-func/src/infra/storage/access-log-register.ts

```diff
@@ -0,0 +1,61 @@
+import { BlockBlobUploadResponse, ContainerClient } from "@azure/storage-blob";
+import {
+  AuditEvent,
+  auditEventSchema,
+} from "io-fims-common/domain/audit-event";
+import * as assert from "node:assert/strict";
+
+// A helper function used to read a Node.js readable stream into a String
+export async function getStreamIntoString(
+  readableStream: NodeJS.ReadableStream,
+) {
+  const chunks = [];
+
+  for await (const chunk of readableStream) {
+    chunks.push(Buffer.from(chunk));
+  }
+
+  return Buffer.concat(chunks).toString("utf-8");
+}
+
+export class AccessLogRegister {
+  #container: ContainerClient;
+
+  constructor(client: ContainerClient) {
+    this.#container = client;
+  }
+
+  async get(name: string): Promise<AuditEvent> {
+    try {
+      const blobClient = this.#container.getBlobClient(name);
+      const downloadBlockBlobResponse = await blobClient.download();
+      const downloadedStream = downloadBlockBlobResponse.readableStreamBody;
+      assert.ok(downloadedStream, `No Blob with name ${name} found`);
```

Comment:
Are we sure that we want to throw an error in this case? If we can't find a blob with this name i don't think there is a point on retry the operation again cc @lucacavallaro

Reply:
The point here is that we want to catch everything possible and return an Error, than it's responsibility of the caller to know what to do with the error

Reply:
Ok but here how can the caller know if this method is throwing an error because of a missing blob? An if on the message it doesn't seems to be a good way to me, in my opinion it is better to avoid throwing an error in this case so that caller can avoid another check to determine whether or not to trigger a retry

Reply:
I get your point, but if the dowloaded file is not present how do you avoid  throwing an error? The method cannot  continue. Maybe we can have a custom error just for that particular case in order to make the caller understand what the problem is with an easy check.

Reply:
Can't we add a check in the catch block based on the type of the error?

Reply:
I can create a custom error and move that part out of the `try-catch` in order to throw a specific error that than the caller can check and decide if a retry is needed or not